### PR TITLE
Add federated daemonset documentation

### PR DESCRIPTION
**Pull request by mwielgus**
cc: @nikhiljindal @madhusudancs

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1827)

### CODE REVIEW

File: docs/user-guide/federation/daemonsets.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+For example, you can do that using kubectl by running:
+
+```shell
+kubectl --context=federation-cluster delete daemonset mydaemonset
+```
+
+Note that at this point, deleting a federated DaemonSet will not delete the
+corresponding DaemonSets from underlying clusters.
+You must delete the underlying DaemonSets manually.
+We intend to fix this in the future.
```
Comment (madhusudancs) [inline docs/user-guide/federation/daemonsets.md]: There is cascading deletion support for Federated DaemonSets now. I think you should point this to - https://github.com/kubernetes/kubernetes.github.io/pull/1787/files#diff-74d1f4f6cca77a8c6f9631bbed28b69eR50 (whatever is the appropriate docs link for that).

**Comment by devin-donnelly** [comment]
Please retarget this PR to the release-1.5 branch if it's a change for 1.5.

**Comment by mwielgus** [comment]
Retargetted.

**Comment by mwielgus** [comment]
@madhusudancs Comments applied.

File: docs/user-guide/federation/daemonsets.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+These DaemonSets in underlying clusters will match the Federated Daemonset.
+
+
+## Updating a Federated Daemonset
+
+You can update a Federated Daemonset as you would update a Kubernetes
+DaemonSet; however, for a Federated Daemonset, you must send the request to
+the federation apiserver instead of sending it to a specific Kubernetes cluster.
+The federation control plan ensures that whenever the Federated Daemonset is
```
Comment (madhusudancs) [inline docs/user-guide/federation/daemonsets.md]: s/plan/plane/

**Comment by madhusudancs** [comment]
@mwielgus one nit.

**Comment by madhusudancs** [comment]
cc @kubernetes/docs

File: docs/user-guide/federation/daemonsets.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+---
+---
+
+This guide explains how to use DaemonSets in Federation control plane.
```
Comment (devin-donnelly) [inline docs/user-guide/federation/daemonsets.md]: "in a Federation control plane" or "with your Federation control plane."

File: docs/user-guide/federation/daemonsets.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes Cluster
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
```
Comment (devin-donnelly) [inline docs/user-guide/federation/daemonsets.md]: Make the link title more descriptive than "this one"; think of it like a variable name.

"Other tutorials, such as Kelsey Hightower's [Federated Kubernetes Tutorial](https://github.com/kelseyhightower/kubernetes-cluster-federation), might also help you create a Federated Kubernetes cluster."

File: docs/user-guide/federation/daemonsets.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+by Kelsey Hightower, are also available to help you.
+
+You are also expected to have a basic
```
Comment (devin-donnelly) [inline docs/user-guide/federation/daemonsets.md]: "You are also expected to have" -> "You should also have"

File: docs/user-guide/federation/daemonsets.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+[working knowledge of Kubernetes](/docs/getting-started-guides/) in
+general and DaemonSets in particular.
+
+## Overview
+
+DaemonSets in federation control plane (referred to as "Federated Daemonsets" in
```
Comment (devin-donnelly) [inline docs/user-guide/federation/daemonsets.md]: Just use "Federated DaemonSets".

"Federated DaemonSets are very similar to..."

File: docs/user-guide/federation/daemonsets.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+this guide) are very similar to the traditional [Kubernetes
+DaemonSets](/docs/user-guide/DaemonSets/) providing the same functionality.
```
Comment (devin-donnelly) [inline docs/user-guide/federation/daemonsets.md]: "providing" -> "and provide"

**Comment by devin-donnelly** [comment]
Basically the same nits as Federated ConfigMaps.

File: docs/user-guide/federation/daemonsets.md (hunk 2 above)
Comment (mwielgus) [inline docs/user-guide/federation/daemonsets.md]: done.

File: docs/user-guide/federation/daemonsets.md (hunk 3 above)
Comment (mwielgus) [inline docs/user-guide/federation/daemonsets.md]: done.

File: docs/user-guide/federation/daemonsets.md (hunk 4 above)
Comment (mwielgus) [inline docs/user-guide/federation/daemonsets.md]: Done.

File: docs/user-guide/federation/daemonsets.md (hunk 6 above)
Comment (mwielgus) [inline docs/user-guide/federation/daemonsets.md]: Done.

File: docs/user-guide/federation/daemonsets.md (hunk 7 above)
Comment (mwielgus) [inline docs/user-guide/federation/daemonsets.md]: done.

**Pull request merged**